### PR TITLE
Trans opt with squeeze

### PR DESCRIPTION
**Pull request by zhijxu-MS**

### CODE REVIEW

File: tf2onnx/onnx_opset/nn.py [hunk 1]
```diff
@@ -481,6 +481,7 @@ def version_7(cls, ctx, node, **kwargs):
         # wants the input to be NHWC - adjust target_shape to this.
         n, h, w, c = shape
         nh, nw = target_shape
+        utils.make_sure(all(i != -1 for i in [nh, nw]), "h and w need to be known")
```
Comment (lucienwang1009) [inline tf2onnx/onnx_opset/nn.py]: please use utils.ONNX_UNKNOWN_DIMENSION

File: tests/test_optimizers.py [hunk 2]
```diff
@@ -49,12 +49,19 @@ def run_and_compare(self, output_names_with_port, onnx_feed_dict, origin_proto,
...
     # Tranpose Optimizer Tests Start
 
     def run_transpose_compare(self, output_names_with_port, onnx_feed_dict, origin_proto,
                               remaining_transpose_num=None, debug=False, rtol=1e-07):
-        self.run_and_compare(output_names_with_port, onnx_feed_dict, origin_proto, op_type="Transpose",
-                             remaining_op_num=remaining_transpose_num, debug=debug, rtol=rtol)
+        return self.run_and_compare(output_names_with_port, onnx_feed_dict, origin_proto, op_type="Transpose",
+                                    remaining_op_num=remaining_transpose_num, debug=debug, rtol=rtol)
+
+    def check_transpose_perm(self, model_proto, expected_perm):
```
Comment (lucienwang1009) [inline tests/test_optimizers.py]: Optional: take it as model_checker option of run_transpose_compare like graph_checker in backend_test_base.py?

File: tf2onnx/onnx_opset/nn.py (hunk 1 above)
Comment (zhijxu-MS) [reply tf2onnx/onnx_opset/nn.py]: converter has the code "utils.ONNX_UNKNOWN_DIMENSION = args.unknown_dim"
if user specifies unknown_dim as 1, then ....

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 3]
```diff
@@ -435,6 +436,45 @@ def _split_handler(self, trans, node):
             return True
         return False
 
+    def _squeeze_handler(self, trans, node):
+        def _calculate_new_attr(ori_perm, ori_squeeze_axes):
+            new_squeeze_axes = sorted([ori_perm[i] for i in ori_squeeze_axes])
+            # calculate output shape after trans and squeeze
+            input_shape = "abcd"
```
Comment (nbcsm) [inline tf2onnx/optimizer/transpose_optimizer.py]: do we have a better way?
[0,1,2,3] is even better than "abcd"...

**Pull request merged**